ffi_how_to.md

```diff
@@ -0,0 +1,115 @@
+This document aims to create a detailed work plan for shipping the `pre-built-ffi` workstream tracked in https://github.com/filecoin-project/filecoin-ffi/issues/209
+The high level approach for this has already been well documented at https://hackmd.io/@mvdan/Hy7iK0TEY.
+
+### Changes to https://github.com/filecoin-project/filecoin-ffi
+
+Going ahead, except for cases where users want to build `filecoin-ffi` from source, `filecoin-ffi` will essentially act as a "wrapper" that forwards all API calls to the corresponding `prebuilt-ffi` module. Since a `prebuilt-ffi` module contains platform-dependent assets (such as static C libraries), APIs in `filecoin-ffi` will need to delegate calls to the "platform-specific" `prebuilt-ffi` module.
+
+Currently, `filecoin-ffi` supports multiple (GOOS + GOARCH) combinations. Therefore, we must modify every public API in `filecoin-ffi` to have a variant for each (GOOS + GOARCH) combination. These variants will delegate the call to the corresponding platform-specific `prebuilt-ffi` module instead of calling into the CGO bindings as they do today (except when users build `filecoin-ffi` from source).
```

Comment:
You don't need separate API for every (GOOS + GOARCH) combination. A shim somewhere (possibly between filecoin-ffi and prebuilt) could use go tags system to perform this delegation.